File: AlienInvaders/app/src/main/java/edu/cs4730/alieninvaders/MainActivity.java
package edu.cs4730.alieninvaders;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // no layout used, since using mySurfaceView
        //requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(new mySurfaceView(this));

    }
}

File: AlienInvaders_kt/app/src/main/java/edu/cs4730/alieninvaders_kt/MainActivity.kt
package edu.cs4730.alieninvaders_kt

import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //with android 15, we need the insets otherwise the surface draws over the things it should not.
        ViewCompat.setOnApplyWindowInsetsListener(
            findViewById(R.id.main)
        ) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            WindowInsetsCompat.CONSUMED
        }
        val myLayout = findViewById<LinearLayout>(R.id.main)
        myLayout.addView(mySurfaceView(this))
    }
}